docs/rfcs/XXXX_storage_backends.md

```diff
@@ -0,0 +1,248 @@
+The default solution for a new organization (and the existing Kids First organization) will be to store all documents within the application bucket. These documents will be kept as outlined above in the `New Layout` where the `studies/` prefix will be at the root of the bucket.
+
+### New Layout
+```
+studies/
+  SD_00000000/
+    source/
+    uploads/
+    harmonized/
+```
+
+## External Storage Backends
+
+![enter image description here](https://i.ibb.co/PTPMLnf/storage-layout.png)
+Under circumstances that an organization requires all of their uploads or data files to be under their control, an organization administrator may specify credentials and details for a bucket of their choosing in a different account.
+
+### Storage Backends
+
+A Storage Backend is a simple entity that belongs to an organization and specifies IAM user credentials, a bucket that the IAM user has access to, and a prefix within that bucket. This concept will allow Dewrangle to access buckets across different accounts for the purpose of storing or evaluating files.
+
+### Configuration
+To configure a storage backend, an organization administrator will be required to provide details about the AWS account where their target bucket resides and a user who may access it. An organization administrator will be able to configure this by navigating to the *Organization Settings* screen and choosing *Storages*. This will allow the administrator to add a new storage by first asking the user to create a new IAM user in their AWS account and assigning a specified policy. After the user has done this, they will be required to provide the AWS access and secret key pair so that Dewrangle may access the bucket. The final step will be to allow the user to specify the bucket and what prefix under it they would like this storage backend to use as the root. After confirming their choices, Dewrangle will attempt to access the bucket with the provided credentials and save the new storage backend if successful.
+
+### Use of Storage Backends
+
+A Storage Backend may be used in one of two ways: Either as a Document Storage Backend or a Data Storage Backend. Each method will require the same configuration of a Storage Backend and, in fact, a single Storage Backend may be used for both Documents and Data.
+
+#### Document Storage Backends
+
+Storage Backends used for documents will be read/write. Dewrangle will use these backends to save documents using a `uploads/` prefix under the root of the Storage Backend.
```

Comment:
Yeah I think this is a good idea to namespace the root. We could do something like `dewrangle/uploads/` as a prefix prepended automatically to everything.

Reply:
We can have `dewrangle` be it's own prefix because it we may want some other contents later in parallel with the uploads. Perhaps we have some extract configs directory or a prefix for holding intermediate results from ETL.